server/etcdserver/api/v3rpc/health.go

```diff
@@ -18,37 +18,47 @@ import (
 	"go.uber.org/zap"
 	"google.golang.org/grpc/health"
 	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+
+	"go.etcd.io/etcd/server/v3/etcdserver"
 )
 
 const (
 	allGRPCServices = ""
 )
 
-type HealthNotifier interface {
-	StartServe()
-	StopServe(reason string)
+type notifier interface {
```

Comment:
Nit: The previous name `HealthNotifier` makes more sense than `notifier`, because it isn't defined inside package "`health`".

Overall this PR is a little over-engineering. Both `(*maintenanceServer) Defragment` and `(*healthChecker) StopServe` are in the same package `api/v3rpc`, we don't have to necessarily introduce or refine the interface for now. 

Anyway, not a big deal for now.

Reply:
On the other side, we should avoid the other extreme. Fact that structs are in single package doesn't mean that they should access all internal methods. Having a internal interface doesn't complicate code too much, but it clarifies the interned usage.